### ISSUE TITLE
RuntimeError: duplicate registrations

### ISSUE THREAD

**Issue by RCheese** [nod-ai/SHARK#2063]
`python apps/shark_studio/web/index.py`
```
/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/transformers/utils/generic.py:441: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
Traceback (most recent call last):
  File "/home/rch/Downloads/SHARK/apps/shark_studio/web/index.py", line 5, in <module>
    from ui.chat import chat_element
  File "/home/rch/Downloads/SHARK/apps/shark_studio/web/ui/chat.py", line 11, in <module>
    from apps.shark_studio.api.llm import (
  File "/home/rch/Downloads/SHARK/apps/shark_studio/api/llm.py", line 1, in <module>
    from turbine_models.custom_models import stateless_llama
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/turbine_models/custom_models/stateless_llama.py", line 9, in <module>
    from shark_turbine.aot import *
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/shark_turbine/aot/__init__.py", line 7, in <module>
    from .compiled_module import CompiledModule
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/shark_turbine/aot/compiled_module.py", line 18, in <module>
    from . import builtins
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/shark_turbine/aot/builtins/__init__.py", line 8, in <module>
    from .jittable import jittable
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/shark_turbine/aot/builtins/jittable.py", line 25, in <module>
    from ...dynamo.passes import (
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/shark_turbine/dynamo/passes.py", line 4, in <module>
    from shark_turbine.dynamo import utils
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/shark_turbine/dynamo/utils.py", line 13, in <module>
    @register_decomposition(torch.ops.aten._scaled_dot_product_flash_attention.default)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/torch/_decomp/__init__.py", line 185, in decomposition_decorator
    pytree.tree_map_(register, aten_op)
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/torch/utils/_pytree.py", line 798, in tree_map_
    deque(map(func, *flat_args), maxlen=0)  # consume and exhaust the iterable
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/torch/_decomp/__init__.py", line 182, in register
    _add_op_to_registry(registry, op, fn)
  File "/home/rch/Downloads/SHARK/shark.venv/lib/python3.11/site-packages/torch/_decomp/__init__.py", line 55, in _add_op_to_registry
    raise RuntimeError(f"duplicate registrations for {op_overload}")
RuntimeError: duplicate registrations for aten._scaled_dot_product_flash_attention.default
```

What could be the problem?

**Comment by dc999999** [comment]
That line was introduced in this PR. 

https://github.com/nod-ai/SHARK-Turbine/pull/294/files/1ba10b7a049f13c29288ca75d1a32085f73036c2#r1437280202

**Comment by monorimet** [comment]
Temporary fix is to pip install torch==2.1.0

@gpetters94 @aviator19941 any updates on a better solution to SDPFA decomposition? If not, we need to make sure the workaround decomposition isn't added if user's torch install has it already

**Comment by dc999999** [comment]
It seems like an if statement should be possible here.
